pkgs/stdenv/generic/check-meta.nix

```
@@ -444,7 +448,14 @@ let
     let
       outputs = attrs.outputs or [ "out" ];
     in
-    {
+    lib.attrsets.optionalAttrs (attrs ? src.meta.homepage) {
+      # should point to an http-browsable source tree, if available.
+      # fetchers like fetchFromGitHub set it automatically.
+      # this could be handled a lot easier if we nulled it instead
+      # of having it be undefined, but that wouldn't match the
+      # other attributes.
+      repository = attrs.src.meta.homepage;
+    } // {
```

this doesn't work with `srcs`. need to handle `srcs` and a mix of fetchers that may or may not have homepages. here's something. can be improved probably.
```suggestion
    lib.attrsets.optionalAttrs (attrs ? src.meta.homepage || attrs ? srcs && any (src: src ? meta.homepage) attrs.srcs) {
      # should point to an http-browsable source tree, if available.
      # fetchers like fetchFromGitHub set it automatically.
      # this could be handled a lot easier if we nulled it instead
      # of having it be undefined, but that wouldn't match the
      # other attributes.
      repository = let
        getSrcs = attrs:
          if attrs ? src
          then
            [ attrs.src ]
          else
            lib.filter (src: src ? meta.homepage) attrs.srcs;
        getHomePages = srcs: map (f: f.meta.homepage) srcs;
        unlist = list:
          if lib.length list == 1
          then
            lib.elemAt list 0
          else
            list;
      in
        unlist (getHomePages (getSrcs attrs));
    } // {
```

---

after a quick search i was surprised to find there seems to be a double-digit number of packages that use `srcs` in conjunction with `fetchFromGitHub`.  i guess i'll tentatively apply this, but if it breaks CI, i'll probably revert.